pkg/execution/execution_test.go

```diff
@@ -0,0 +1,406 @@
+		var exec = require('k6/x/execution');
+
+		exports.default = function() {
+			var vuStats = exec.getVUStats();
+			if (vuStats.id !== 1) throw new Error('unexpected VU ID: '+vuStats.id);
+			if (vuStats.idGlobal !== 10) throw new Error('unexpected global VU ID: '+vuStats.idGlobal);
+			if (vuStats.iteration !== 0) throw new Error('unexpected VU iteration: '+vuStats.iteration);
+			if (vuStats.iterationScenario !== 0) throw new Error('unexpected scenario iteration: '+vuStats.iterationScenario);
+		}`},
+		{name: "vu_err", script: `
+		var exec = require('k6/x/execution');
+		exec.getVUStats();
+		`, expErr: "getting VU information in the init context is not supported"},
+		{name: "scenario_ok", script: `
+		var exec = require('k6/x/execution');
+		var sleep = require('k6').sleep;
+
+		exports.default = function() {
+			var ss = exec.getScenarioStats();
+			sleep(0.1);
+			if (ss.name !== 'default') throw new Error('unexpected scenario name: '+ss.name);
+			if (ss.executor !== 'test-exec') throw new Error('unexpected executor: '+ss.executor);
+			if (ss.startTime > new Date().getTime()) throw new Error('unexpected startTime: '+ss.startTime);
+			if (ss.progress !== 0.1) throw new Error('unexpected progress: '+ss.progress);
+			if (ss.iteration !== 3) throw new Error('unexpected scenario local iteration: '+ss.iteration);
+			if (ss.iterationGlobal !== 4) throw new Error('unexpected scenario local iteration: '+ss.iterationGlobal);
+		}`},
+		{name: "scenario_err", script: `
+		var exec = require('k6/x/execution');
+		exec.getScenarioStats();
+		`, expErr: "getting scenario information in the init context is not supported"},
+		{name: "test_ok", script: `
+		var exec = require('k6/x/execution');
+
+		exports.default = function() {
+			var ts = exec.getTestInstanceStats();
+			if (ts.duration !== 0) throw new Error('unexpected test duration: '+ts.duration);
+			if (ts.vusActive !== 1) throw new Error('unexpected vusActive: '+ts.vusActive);
+			if (ts.vusMax !== 0) throw new Error('unexpected vusMax: '+ts.vusMax);
+			if (ts.iterationsCompleted !== 0) throw new Error('unexpected iterationsCompleted: '+ts.iterationsCompleted);
+			if (ts.iterationsInterrupted !== 0) throw new Error('unexpected iterationsInterrupted: '+ts.iterationsInterrupted);
+		}`},
+		{name: "test_err", script: `
+		var exec = require('k6/x/execution');
+		exec.getTestInstanceStats();
+		`, expErr: "getting test information in the init context is not supported"},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			r, err := getSimpleRunner(t, "/script.js", tc.script)
+			if tc.expErr != "" {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tc.expErr)
+				return
+			}
+			require.NoError(t, err)
+
+			samples := make(chan stats.SampleContainer, 100)
+			initVU, err := r.NewVU(1, 10, samples)
+			require.NoError(t, err)
+
+			execScheduler, err := local.NewExecutionScheduler(r, testutils.NewLogger(t))
```

Comment:
Should we simplify these into unit tests, and avoid the `ExecutionScheduler` and the utils copied to make it work?

Reply:
I doubt this repo will long enough for this to matter all that much